cmd/arcaflow/main.go

```diff
@@ -57,55 +58,33 @@ func main() {
 	workflowFile := "workflow.yaml"
 	printVersion := false
 
-	flag.BoolVar(&printVersion, "version", printVersion, "Print Arcaflow Engine version and exit.")
-	flag.StringVar(
-		&configFile,
-		"config",
-		configFile,
-		"The Arcaflow configuration file to load, if any.",
+	const (
+		helpUsage    = "Display this help output."
+		versionUsage = "Print Arcaflow Engine version and exit."
+		configUsage  = "The Arcaflow configuration file to load, if any."
+		inputUsage   = "The workflow input file to load. May be outside the workflow " +
```

Comment:
It seems that all paths are *expected* to be absolute, or otherwise relative to the current working directory. It seems a little strange to me that the paths are not relative to the context directory. I'm not even sure what the point of the context directory is..

Reply:
> It seems a little strange to me that the paths are not relative to the context directory. I'm not even sure what the point of the context directory is..

You're right; and that's interesting, and arguably wrong. There's this `dir = "."` default, which is passed as the root to the file cache loader. I kinda suspect that ought to have been replaced by the `--context` value, if specified, so resolve relative paths relative to the context directory. And, if not, indeed it would seem that the purpose of `--context` is solely (as Jay Leno once said about turn signals in Boston) "to annoy and confuse".

Reply:
@dbutenhof You've made me realize, and I confirmed with testing, that the code actually _does_ use the context directory when looking for the config file, but not for the input or workflow files. For instance, this works:

Reply:
This raises an interesting question:  is the "input" part of the "context" or is it separate from the context?

If you were running the workflow repeatedly with the same input, then you would likely be surprised if the input file were not handled in the same way as the other "inputs" (i.e., the workflow and the configuration); however, if your MO is to run the same workflow with lots of different inputs, then it might seem obvious to you that the input file is _not_ part of the "context" (which contains the files which **don't** vary from run to run).

So...I don't know which approach is most likely to conform to the principle of least surprise.

Looking at the modes of failure, I suspect that in the multi-input-file use case the input file is more likely to be specified with a full path for the file, and so having the input file location default to the context directory is probably the least surprising.  (It would be easy enough to specify `./input.yaml` to anchor it to the current working directory (although, I don't know if that's enough to prompt the engine to look for it there instead of relative to the context directory...perhaps `$PWD/input.yaml` is required...).)

Reply:
> It would be easy enough to specify `./input.yaml` to anchor it to the current working directory (although, I don't know if that's enough to prompt the engine to look for it there instead of relative to the context directory...perhaps `$PWD/input.yaml` is required...).

After I posted this, I ended up looking at the code and I believe that `./input.yaml` would be interpreted as a relative path and would be located relative to the context directory, so the user would need to use `$PWD` rather than a dot, assuming that the invocation is made under a shell.